deploy/kubernetes/releases/csi-digitalocean-dev/driver.yaml

```diff
@@ -405,6 +405,11 @@ spec:
               # needed so that any mounts setup inside this container are
               # propagated back to the host machine.
               mountPropagation: "Bidirectional"
+            - name: kubelet-data-dir
+              mountPath: /var/data/kubelet
```

Comment:
Is this to support a kubelet root directory alternative to the default `/var/lib/kubelet` (i.e., your deployment of kubelet diverges from the default)?

Reply:
@timoreimann yeah if that’s ok.

Reply:
I can totally see how you (and others) may want to use a non-standard directory. What I'm not too sure about is whether adding additional mount points is the right way to achieve that. At the least, it could be surprising/confusing to users to see several mounts where only one is effectively used.

My suggestion would be to keep the manifests as-is and adjust your local copy to your needs. It's also possible to use one of the many manifest management tools out there to keep upstream (i.e., our manifests) unchanged and put customizations on top in a maintainable fashion. Examples are kustomize, kpt, and ytt.

Let me know if that works for you.

Reply:
If we want something where the user must override then we must have a supported deployment mechanism that can be templatized or passed based on the environment being deployed. Ideally, these kustomizations or whatever method should be kept at the upstream directory of digital ocean

Reply:
We should agree on that tooling first

Reply:
Alternatively, an operator is the best option but that is more work

Reply:
Right, the approach to customize depends a lot on the tool. For instance, kustomize and kpt can operate on upstream resources that don't support customization whatsoever, whereas Helm charts are (to my knowledge) usually bundled with the upstream source. 

We also had an external contributor starting the work to support Helm (#382) but unfortunately couldn't carry the effort onward anymore due to time constraints. We'd be happy to see other contributors step up.